packages/common/src/utils/urlFormatting.test.ts

```diff
@@ -0,0 +1,14 @@
+import { createURL } from './urlFormatting';
+
+describe('createUrl', () => {
+  test('valid url from a path, query params', async () => {
+    const url = createURL('/test', { foo: 'bar' });
+
+    expect(url).toEqual('/test?foo=bar');
+  });
+  test('valid url from a path including params, query params', async () => {
+    const url = createURL('/test?existing-param=1', { foo: 'bar' });
+
+    expect(url).toEqual('/test?existing-param=1&foo=bar');
+  });
```

Comment:
Great that you also wrote some tests for it. Maybe also add a test for the removal from a queryParam?

```suggestion
  test('valid url from a path including params, query params', async () => {
    const url = createURL('/test?existing-param=1', { foo: 'bar' });

    expect(url).toEqual('/test?existing-param=1&foo=bar');
  });
 test('valid url from a path including params, query params', async () => {
    const url = createURL('/test?foo=1&bar=1', { foo: null});

    expect(url).toEqual('/test?bar=1');
  });
```

Reply:
Good idea! Added it.